### DISCUSSION TITLE
can you add knex query builder ?

### DISCUSSION THREAD

**Discussion by applemate** [prisma/prisma#24093] [Q&A]
can you add knex query builder ?

would be nice if you can do this

```js
db.select().from(db.products).where({id:3})
```

**Comment by janpio** [comment]
What is `db` here? 
And what would this do in context of Prisma?

**Reply by applemate** [reply]
`db` could be `prisma` or anything that works

**Reply by janpio** [reply]
Then the answer is that you could do that yourself: Use Knex to generate the SQL query and then hand that over to Prisma as a [raw query](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries) to execute.

Right now we do not plan to _integrate_ Knex in any way.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
